src/Services/Catalog/Catalog.API/Extensions/RedisCacheExtensions.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.IO;
+
+using System.Threading.Tasks;
+using System.Runtime.Serialization.Formatters.Binary;
+
+using Microsoft.Extensions.Caching.Distributed;
+using System.Collections.Generic;
+
+namespace Catalog.API.Extensions
+{
+    public static class RedisCacheExtensions
+    {
+        public static readonly string KEYS = "KEYS";
+
+        public static async Task<T> TryGetAsync<T>(this IDistributedCache cache, string key)
+        {
+            var value = await cache.GetAsync(key);
+
+            if (value != null)
+            {
+                return FromByteArray<T>(value);
+            }
+            return default(T);
+        }
+
+        public static async Task TrySetAsync<T>(this IDistributedCache cache, string key, T value)
+        {
+            if (value == null)
+                return;
+
+            T temp = await cache.TryGetAsync<T>(key);
+
+            if (temp == null || temp.Equals(default(T)))
+            {
+                var array = ToByteArray<T>(value);
+                cache.SetKeysAsync(key);
+                await cache.SetAsync(key, array);
+            }
+        }
+
+        public static async Task<bool> TryResetAsync<T>(this IDistributedCache cache, string key, T value)
+        {
+            T temp = await cache.TryGetAsync<T>(key);
+
+            if (temp != null && !temp.Equals(default(T)))
+            {
+                await cache.RemoveAsync(key);
+            }
+
+            var array = ToByteArray<T>(value);
+            await cache.TrySetAsync(key, value);
+
+            return true;
+        }
+
+        public static async Task<bool> RemovePartialAsync<T>(this IDistributedCache cache, List<string> dismissList)
+        {
+            var values = await cache.TryGetAsync<List<string>>(KEYS);
+            return await Task.Run(async () => {
+                values.ForEach(async k =>
+                {
+                    if (!dismissList.Contains(k))
+                    {
+                        T temp = await cache.TryGetAsync<T>(k);
+
+                        if (temp != null && !temp.Equals(default(T)))
+                        {
+                            await cache.RemoveAsync(k);
+                        }
+                    }
+                });
+                await cache.RemoveAsync(KEYS);
+                return true;
+            });            
+        }
+
+        private static byte[] ToByteArray<T>(T obj)
+        {
+            if (obj == null)
+                return null;
+
+            BinaryFormatter bf = new BinaryFormatter();
```

Comment:
A better name for bf and ms.

Reply:
bf will be renamed as VarBinaryFormatter and ms as VarMemoryStream